application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,311 @@
+        ZonedDateTime lastSavedDate = getLocalDateTime(dateStr, feedConfig.dateFormatterPattern());
+
+        final Predicate<Item> shouldItemBePosted = item -> {
+            ZonedDateTime itemPubDate =
+                    getDateTimeFromItem(item, feedConfig.dateFormatterPattern());
+            return itemPubDate.isAfter(lastSavedDate);
+        };
+
+        // Date that will be stored in the database at the end
+        ZonedDateTime lastPostedDate = lastSavedDate;
+
+        // Send each item that should be posted and concurrently
+        // find the post with the latest date
+        for (Item item : rssItems.reversed()) {
+            if (!shouldItemBePosted.test(item)) {
+                continue;
+            }
+
+            MessageEmbed embed = constructEmbedMessage(item, feedConfig).build();
+            textChannel.sendMessageEmbeds(List.of(embed)).queue();
+
+            // Get the last posted date so that we update the database
+            ZonedDateTime pubDate = getDateTimeFromItem(item, feedConfig.dateFormatterPattern());
+            if (pubDate.isAfter(lastPostedDate)) {
+                lastPostedDate = pubDate;
+            }
+        }
+
+        // Finally, save the last posted date to the database.
+        DateTimeFormatter dateTimeFormatter =
+                DateTimeFormatter.ofPattern(feedConfig.dateFormatterPattern());
+        String lastDateStr = lastPostedDate.format(dateTimeFormatter);
+        if (dateResult == null) {
+            database.write(context -> context.newRecord(RSS_FEED)
+                .setUrl(feedConfig.url())
+                .setLastDate(lastDateStr)
+                .insert());
+            return;
+        }
+
+        // If we already have an existing record with the given URL,
+        // now is the time to update it
+        database.write(context -> context.update(RSS_FEED)
+            .set(RSS_FEED.LAST_DATE, lastDateStr)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .executeAsync());
+    }
+
+    /**
+     * Attempts to get a {@link ZonedDateTime} from an {@link Item} with a provided string date time
+     * format.
+     * <p>
+     * If either of the function inputs are null, the oldest-possible {@link ZonedDateTime} will get
+     * returned instead.
+     *
+     * @return The computed {@link ZonedDateTime}
+     */
+    private static ZonedDateTime getDateTimeFromItem(Item item, String dateTimeFormat) {
+        String pubDate = item.getPubDate().orElse(null);
+
+        if (pubDate == null || dateTimeFormat == null) {
+            return ZONED_TIME_MIN;
+        }
+
+        return getLocalDateTime(pubDate, dateTimeFormat);
+    }
+
+    /**
+     * Given a list of RSS feed items, this function checks if the dates are valid
+     * <p>
+     * This assumes that all items share the same date format (as is usual in an RSS feed response),
+     * therefore it only checks the first item's date for the final result.
+     */
```

Comment:
Same here, do the doc properly and remove the p, and also, you are missing a `.` at the end of the first line.

Reply:
Added the dot! I've talked about the `<p>` tag in a previous comment :)